Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml

```diff
@@ -27,7 +62,25 @@ ParserQuery: |
   ];
   // -- Common preprocessing to both input and outbound events
   let RawNetworkEvents =
-    DeviceNetworkEvents
+    DeviceNetworkEvents | where not(disabled)
+    // ***************  Prefilterring *****************************************************************
+    |where (isnull(starttime) or TimeGenerated>=starttime) and
+           (isnull(endtime) or TimeGenerated<=endtime) and
+           (isnull(dstportnumber) or dstportnumber == LocalPort or dstportnumber == RemotePort) and
+           (array_length(url_has_any)==0  or RemoteUrl has_any (url_has_any)) and
+           (array_length(srcipaddr_has_any_ipv4_prefix)==0 
+              or has_any_ipv4_prefix(RemoteIP,srcipaddr_has_any_ipv4_prefix)
+              or has_any_ipv4_prefix(LocalIP ,srcipaddr_has_any_ipv4_prefix)
+            ) and
+           (array_length(dstipaddr_has_any_ipv4_prefix)==0 
+              or has_any_ipv4_prefix(RemoteIP,dstipaddr_has_any_ipv4_prefix)
+              or has_any_ipv4_prefix(LocalIP ,dstipaddr_has_any_ipv4_prefix)
+            ) and
+           (array_length(hostname_has_any)==0 
+              or RemoteUrl has_any(hostname_has_any) or DeviceName has_any(hostname_has_any) ) 
+           // no ua data: (isnull(httpuseragent_has_any) or ) and
```

Comment:
We actually need to check that the parameter is at default and otherwise return nothing.